### PR TITLE
fix(react): fix component composition issues

### PR DESCRIPTION
**Pull request by brionmario**
### Purpose

In this PR:
1. Removed the usage of `defaultProps` on functional components and refactored to use ES6 syntax. (See: https://github.com/facebook/react/pull/16210)
2. Fix composition issues due to:
 - Not forwarding refs properly
 - Typing issues when using `component` prop.

### Related Issues
- Fixes https://github.com/wso2/oxygen-ui/issues/277
- Fixes https://github.com/wso2/oxygen-ui/issues/282
- Fixes https://github.com/wso2/oxygen-ui/issues/276

### Related PRs
- N/A

### Checklist
- [ ] Figma Board Updated. (Mandatory for Icon and Component PRs. If you don't have access, please ask the core team to update it.)
- [ ] UX/UI review done on the final implementation.
- [ ] Story provided. (Add screenshots)
- [ ] Manual test round performed and verified.
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Documentation provided. (Add links if there are any)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [x] Ran ESLint & Prettier plugins and verified?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: packages/react/src/components/CountryFlag/CountryFlag.tsx [hunk 1]
```diff
@@ -31,20 +32,45 @@ export interface CountryFlagsProps extends React.HTMLAttributes<HTMLElement & SV
...
    */
   height?: string;
-}
+} & HTMLAttributes<HTMLImageElement>;
 
-const COMPONENT_NAME: string = 'Flag';
+/**
+ * @deprecated Use the {@link CountryFlagProps} instead.
+ */
+export type CountryFlagsProps = CountryFlagProps;
```
Comment (brionmario) [inline packages/react/src/components/CountryFlag/CountryFlag.tsx:40]: https://github.com/wso2/oxygen-ui/issues/285#issuecomment-2406687849

File: packages/react/src/components/Accordion/Accordion.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
+ * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com). All Rights Reserved.
```
Comment (pavinduLakshan) [inline packages/react/src/components/Accordion/Accordion.tsx:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (https://www.wso2.com).
```

**Review by pavinduLakshan — APPROVED** [review]

File: packages/react/src/components/ActionCard/ActionCard.tsx [hunk 3]
```diff
@@ -48,33 +51,57 @@ export interface ActionCardProps extends CardProps {
...
+};
 
 const COMPONENT_NAME: string = 'ActionCard';
 
-const ActionCard: FC<ActionCardProps> & WithWrapperProps = (props: ActionCardProps): ReactElement => {
-  const {className, image, title, description, actionText, onActionClick, ...rest} = props;
-
-  const classes: string = clsx('oxygen-action-card', className);
+/**
+ * The Action Card component is an extended version of the` Card` component with an action button.
```
Comment (savindi7) [inline packages/react/src/components/ActionCard/ActionCard.tsx:59]:
```suggestion
 * The Action Card component is an extended version of the `Card` component with an action button.
```

**Review by pavinduLakshan — APPROVED** [review]

**Review by pavinduLakshan — DISMISSED** [review]

File: packages/react/src/components/Alert/Alert.tsx [hunk 4]
```diff
@@ -16,29 +16,61 @@
...
+} & Omit<MuiAlertProps, 'component'>;
 
 const COMPONENT_NAME: string = 'Alert';
 
-const Alert: ForwardRefExoticComponent<AlertProps> & WithWrapperProps = forwardRef(
-  (props: AlertProps, ref: MutableRefObject<HTMLDivElement>): ReactElement => {
-    const {className, ...rest} = props;
-
+/**
+ * The Alert component display brief messages for the user without interrupting their use of the app.
```
Comment (savindi7) [inline packages/react/src/components/Alert/Alert.tsx:40]:
```suggestion
 * The Alert component displays brief messages to the user without interrupting their use of the app.
```

File: packages/react/src/components/Alert/Alert.tsx [hunk 5]
```diff
@@ -16,29 +16,61 @@
...
+ *
+ * Demos:
+ *
+ * - [Alert (Oxygen UI)] (https://wso2.github.io/oxygen-ui/react/?path=/docs/feedback-alert)
```
Comment (savindi7) [inline packages/react/src/components/Alert/Alert.tsx:44]:
```suggestion
 * - [Alert (Oxygen UI)](https://wso2.github.io/oxygen-ui/react/?path=/docs/feedback-alert)
```

Comment (savindi7) [reply packages/react/src/components/Alert/Alert.tsx:44]: The space is there in other places as well..

Comment (brionmario) [reply packages/react/src/components/Alert/Alert.tsx]: There were several other places with this same problem.

Fixed all with https://github.com/wso2/oxygen-ui/pull/281/commits/262b1c99732e3a4da94f6b9f200aecaa8e09df74

File: packages/react/src/components/Alert/Alert.tsx (hunk 4 above)
Comment (brionmario) [reply packages/react/src/components/Alert/Alert.tsx]: Fixed with https://github.com/wso2/oxygen-ui/pull/281/commits/262b1c99732e3a4da94f6b9f200aecaa8e09df74

File: packages/react/src/components/ActionCard/ActionCard.tsx (hunk 3 above)
Comment (brionmario) [reply packages/react/src/components/ActionCard/ActionCard.tsx]: Fixed with https://github.com/wso2/oxygen-ui/pull/281/commits/262b1c99732e3a4da94f6b9f200aecaa8e09df74

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**